app/graphql/resolvers/concerns/resolvers/submissionable.rb

```diff
@@ -0,0 +1,45 @@
+# Include this module in any of Resolvers if you want to perform
+# a submission lookup and arguments validation based on
+# submission id/external_id.
+
+# Lookup is performed by @arguments[:id]/@arguments[:external_id] by
+# default, be can be overwritten be redefining `submission_id` and
+# `external_submission_id` in your resolvers.
+
+module Resolvers::Submissionable
+  IdsNotPassed =
+    GraphQL::ExecutionError.new('Neither id nor externalId have been passed.')
+
+  def submission
+    @submission ||=
+      Submission.find_by(id: submission_id) ||
+        Submission.find_by(uuid: external_submission_id)
```

Comment:
**non-blocking**: maybe it makes sense to change order and look for submissions by UUID first.

Reply:
Hmm, just realized it might be a blocker since UUID will be cast to a number.

Reply:
@kiryl-zubarau in this case there will be no such problem, since GraphQL clients differentiate `submission_id` and `external_submission_id` on their side. So it shouldn't be the case when force sends a query or mutation with `submission(id: "<uuid>")`

Reply:
Missed that, thanks!